### PR TITLE
Users crud and Users list view changes

### PR DESCRIPTION
**Pull request by carlosbflores**
- Changes in users view for creating users
- Add users paginated list view

### CODE REVIEW

File: src/app/users/routes.js [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
     appRoutes: {
-      users: 'showUsers'
+      'users(/)': 'showUsers',
+      'users/new': 'newUser',
     },
     controller: {
+      //paginated users
       showUsers() {
-        app.showViewOnRoute(new UsersView());
+        if (app.getSession().userHasRole('ROLE_ROOT')
```
Comment (rodrigojv) [inline src/app/users/routes.js]: This if statement to check the ROLE should not be here. 

The users menu option should be enable by default, and should be hidden for users with role:
* ROLE_SURVEY_USER (hidden)
* ROLE_APP_ADMIN (hidden)
* ROLE_USER (hidden)

We should hide or show the menu options when be build the header, this is done in `headers/storage`.

File: src/app/users/storage.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import Storage from 'backbone.storage';
+import Model from './model';
+import Collection from './collection';
+
+var UsersStorage = Storage.extend({
+  model: Model,
+  collection: Collection,
+  getSubHeaderItems(model) {
```
Comment (rodrigojv) [inline src/app/users/storage.js]: I believe this method is not used and therefore should be removed.

File: src/app/users/add/view.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+          $('#select-role').append(
+            $('<option></option>')
+              .attr('value', element.id)
+              .text(element.role)
+          );
+        });
+      }
+    });
+  },
+  loadOrgs() {
```
[the remark this answers is not in the document]
Comment (carlosbflores) [reply src/app/users/add/view.js]: Method simplified, Fetch and Sellect creation repeated code moved to a method to be reusable. Simplified calls to backend, less calls in some cases.

File: src/app/users/routes.js (hunk 1 above)
Comment (carlosbflores) [reply src/app/users/routes.js]: Role check condition removed, menu option shown/hidden in header depending on the role logged in.

File: src/app/users/storage.js (hunk 2 above)
Comment (carlosbflores) [reply src/app/users/storage.js]: Method removed

File: src/app/users/add/view.js [hunk 4]
```diff
@@ -24,135 +25,140 @@ export default Mn.View.extend({
...
+      //  List this Hub's organizations
+      let hubId = session.get('user').application.id;
+      this.genericFetch(
+        `/applications/${hubId}/organizations`,
+        '#select-org',
+        'Select Organization',
+        'organization',
+        'name'
+      );
+    } else if (session.userHasRole('ROLE_APP_ADMIN')) {
```
Comment (rodrigojv) [inline src/app/users/add/view.js]: If the user is app_admin we don't need to show the select.

The main goal should be to have as few if statements as possible, so we have two preferable options here:

1. Don't show the select and add this logic on the backend.
2. Don't show the select and Implement this logic before saving the model. If the model does not have any organization or application set, add the current user organization and application to the model.

Comment (carlosbflores) [reply src/app/users/add/view.js]: Changed user form, if App Admin, hide Organization/Partner select, get organization/application from logged admin and set before saving.

**Pull request merged**